docs: 📄 added page for dataset collections

slugb0t opened the pull request:
Collections page added to documentation page in manage datasets section.

aaronm-2112 reviewed (changes requested):
For the How To section can you change step two so that it says Add / delete collection names.
It is a little confusing as is. 

To go an extra step maybe it is best to make a h3 for describing how to add collection names and a separate h3 for describing how to delete them.

slugb0t commented:
> For the How To section can you change step two so that it says Add / delete collection names. It is a little confusing as is.
> 
> To go an extra step maybe it is best to make a h3 for describing how to add collection names and a separate h3 for describing how to delete them.

Updated based on feedback and also added details about dataset collections can only be modified by owner or manager.

aaronm-2112 reviewed (approved):
Looks good. We can merge this after the 8.0.0 release.
